modules/ingest-geoip/src/test/java/org/elasticsearch/ingest/geoip/direct/DatabaseConfigurationTests.java

```diff
@@ -35,26 +44,39 @@ protected DatabaseConfiguration createTestInstance() {
     }
 
     public static DatabaseConfiguration randomDatabaseConfiguration(String id) {
-        return new DatabaseConfiguration(id, randomFrom(MAXMIND_NAMES), new Maxmind(randomAlphaOfLength(5)));
+        DatabaseConfiguration.Provider provider = switch (between(0, 2)) {
+            case 0 -> new Maxmind(randomAlphaOfLength(5));
+            case 1 -> new Web();
+            case 2 -> new Local(randomAlphaOfLength(10));
+            default -> throw new AssertionError("failure, got illegal switch case");
+        };
+        return new DatabaseConfiguration(id, randomFrom(MAXMIND_NAMES), provider);
     }
 
     @Override
     protected DatabaseConfiguration mutateInstance(DatabaseConfiguration instance) {
         switch (between(0, 2)) {
             case 0:
-                return new DatabaseConfiguration(instance.id() + randomAlphaOfLength(2), instance.name(), instance.maxmind());
+                return new DatabaseConfiguration(instance.id() + randomAlphaOfLength(2), instance.name(), instance.provider());
             case 1:
                 return new DatabaseConfiguration(
                     instance.id(),
                     randomValueOtherThan(instance.name(), () -> randomFrom(MAXMIND_NAMES)),
-                    instance.maxmind()
+                    instance.provider()
                 );
             case 2:
-                return new DatabaseConfiguration(
-                    instance.id(),
-                    instance.name(),
-                    new Maxmind(instance.maxmind().accountId() + randomAlphaOfLength(2))
-                );
+                DatabaseConfiguration.Provider provider = instance.provider();
+                DatabaseConfiguration.Provider modifiedProvider;
+                if (provider instanceof Maxmind maxmind) {
```

Comment:
Would switch pattern matching be cleaner here? I think we are targeting > Java 21?

Reply:
We've got to backport this to 8.x, which is on Java 17. I don't think switching by type is supported in 17 (I'll double-check).